### PR TITLE
Project sanitization + Continuous integration

### PR DESCRIPTION
**Pull request by Julien00859**
Hello :)

Last step from #6 was an invitation to setup a continuous integration. You explained me you didn't know much about CI stuff so I decided to do it myself. Here is a few commits that sanitize your project a bit (don't worry, there are just *very limited* stuff, like trailing whitespace removal) and roll-in a github action that will automatically test new changes.

Please read every commit message :)

Also here is a few documentation you might be interested in:

* https://help.github.com/en/actions (what are github actions, how to use them)
* https://pre-commit.com/ (how to setup the pre-commit hook locally)
* https://flake8.pycqa.org/en/latest/
* https://github.com/pre-commit/pre-commit-hooks
* https://www.git-scm.com/docs/gitattributes (just the CRLF section)

Also I forgot to include the step to build the documentation but I think it would be a nice if you do it yourself so you get into that actions stuff :P

### CODE REVIEW

File: tests/test_Bot.py [hunk 1]
```diff
@@ -3,12 +3,14 @@
...
-with open("calls.json","r") as f:
-	calls = json.load(f)
 
-with open("responses.json","r") as f:
-	responses = json.load(f)
+root = pathlib.Path(__file__).resolve().parent
+with root.joinpath("calls.json").open() as f1, \
+     root.joinpath("responses.json").open() as f2:
+	calls = json.load(f1)
+	responses = json.load(f2)
```
Comment (Julien00859) [inline tests/test_Bot.py:13]: Had to change that path so we can execute the tests from anywhere

File: piscord/Gateway.py [hunk 2]
```diff
@@ -85,8 +86,9 @@ async def _reconnect(self):
 			"d": {
 				"token": self.token,
 				"session_id": self.session_id,
-				"seq": self.last_sequence
-		}}
+				"seq": self.last_sequence,
+   },
+  }
```
Comment (Julien00859) [inline piscord/Gateway.py:91]: Bruh... the closing brackets are not well indented :(

**Comment by Astremy** [comment]
Hello :)

Thank you for your PR, it helps with a few things, but could you fix two problems?

1. The .gitignore: It contains a lot of useless stuff.
Here is the content of the gitignore I made (which I hadn't committed yet)
```
# Pycache
__pycache__/

# Package
build/
dist/
*.egg-info/

# Unit test
.pytest_cache/

# Sphinx documentation
docs/build/

# Test File
test.py
```

2. The requirements.txt: 
Leave only the useful things

Thanks you

**Comment by Julien00859** [comment]
Hello, dumping it from ancient time long forget from everyone

What do you think we just merge https://github.com/Astremy/Piscord/pull/7/commits/3ef74dac2cf29152d277f666953739d3bfb66f6c ?

**Pull request closed without merging**